python/shark_turbine/dynamo/importer.py

```diff
@@ -497,6 +509,35 @@ def _import_default_value(self, loc: Location, arg, expected_jit_type) -> Value:
         # torch.dtype, torch.device, torch.memory_format, torch.layout
         # list
 
+    def _import_enum_value(self, loc: Location, arg):
+        """Import the correct integer representation of torch enum values"""
+        assert (
+            type(arg) in TORCH_ENUM_TYPES
+        ), f"Expected enum type to be one of {TORCH_ENUM_TYPES}, got {type(arg)}"
+        mapping = None
+        enum_cls = None
+
+        if isinstance(arg, torch.dtype):
```

Comment:
Nit: the TypeSubclassMap below is intended to be a more efficient way to manage such cascaded isinstance conditions (by turning them from a cascade of expensive lookups to a direct hash lookup that can be specified in a module level global vs in code.

Have a look and fine to adapt in a followup.

Reply:
Good point, I thought to keep them logically separated but that makes sense. I've pushed a commit refactoring it to use the TypeSubclassMap